p2p/net/mock/mock_test.go

```diff
@@ -319,28 +318,35 @@ func TestStreams(t *testing.T) {
 
 }
 
-func makePinger(st string, n int) func(network.Stream) {
-	return func(s network.Stream) {
+func makePinger(t *testing.T, st string, n int) func(network.Stream) chan struct{} {
+	t.Helper()
+
+	return func(s network.Stream) chan struct{} {
+		done := make(chan struct{}, 1)
 		go func() {
+			defer func() { done <- struct{}{} }()
 			defer helpers.FullClose(s)
 
 			for i := 0; i < n; i++ {
 				b := make([]byte, 4+len(st))
 				if _, err := s.Write([]byte("ping" + st)); err != nil {
-					panic(err)
+					t.Fatal(err)
```

Comment:
nit: go doesn't like calling fatal from a goroutine. We should probably just Error and return.

Reply:
Fixed.